cpp/src/converters/renumber.cu

```diff
@@ -34,79 +44,176 @@ gdf_error gdf_renumber_vertices(const gdf_column *src, const gdf_column *dst,
-
-    ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
-    gdf_column_view(src_renumbered, tmp, src->valid, src->size, src->dtype);
-
-    ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
-    gdf_column_view(dst_renumbered, tmp, dst->valid, dst->size, dst->dtype);
-
-    gdf_error err = cugraph::renumber_vertices(src_size,
-                                               (const int32_t *) src->data,
-                                               (const int32_t *) dst->data,
-                                               (int32_t *) src_renumbered->data,
-                                               (int32_t *) dst_renumbered->data,
-                                               &new_size, &tmp);
-    if (err != GDF_SUCCESS)
-      return err;
-
-    gdf_column_view(numbering_map, tmp, nullptr, new_size, src->dtype);
-  } else if (src->dtype == GDF_INT64) {
-
-    //
-    //  NOTE: At the moment, we force the renumbered graph to use
-    //        32-bit integer ids.  Since renumbering is going to make
-    //        the vertex range dense, this limits us to 2 billion
-    //        vertices.
-    //
-    //        The renumbering code supports 64-bit integer generation
-    //        so we can run this with int64_t output if desired...
-    //        but none of the algorithms support that.
-    //
-    int64_t *tmp;
-    ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
-    gdf_column_view(src_renumbered, tmp, src->valid, src->size, GDF_INT32);
-
-    ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
-    gdf_column_view(dst_renumbered, tmp, dst->valid, dst->size, GDF_INT32);
-
-    gdf_error err = cugraph::renumber_vertices(src_size,
-                                               (const int64_t *) src->data,
-                                               (const int64_t *) dst->data,
-                                               (int32_t *) src_renumbered->data,
-                                               (int32_t *) dst_renumbered->data,
-                                               &new_size, &tmp);
-    if (err != GDF_SUCCESS)
-      return err;
-
-    //
-    //  If there are too many vertices then the renumbering overflows so we'll
-    //  return an error.
-    //
-    if (new_size > 0x7fffffff) {
-      ALLOC_FREE_TRY(src_renumbered, stream);
-      ALLOC_FREE_TRY(dst_renumbered, stream);
-      return GDF_COLUMN_SIZE_TOO_BIG;
+  switch (src->dtype) {
+  case GDF_INT32:
+    {
+      size_t new_size;
+      int32_t *tmp;
+
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(src_renumbered, tmp, src->valid, src->size, src->dtype);
+
+      ALLOC_TRY((void**) &tmp, sizeof(int32_t) * src->size, stream);
+      gdf_column_view(dst_renumbered, tmp, dst->valid, dst->size, dst->dtype);
+
+      gdf_error err = cugraph::renumber_vertices(src->size,
+                                                 (const int32_t *) src->data,
```

Comment:
I recommend using C++ style type casting (static_cast in general and reinterpret_cast only when it is absolutely necessary, and if we need to use reinterpret_cast, then we may add comment to explain why it is OK), static_cast is safer than C style casting which works like reinterpret_cast.